src/docs/user/references/Variables.adoc

```diff
@@ -0,0 +1,408 @@
+
+[source, groovy]
+----
+println variables.get("PA_JOB_ID") // will display the id of the current job
+----
+
+From a native executable, for instance a Shell script, variables are accessible through environment variables.
+
+[source, bash]
+----
+#!/bin/bash
+echo $variables_PA_TASK_NAME
+----
+
+WARNING: In native tasks and Bash/CMD script engines, variables can be read but not written to.
+
+The `variables` map is propagated down the execution chain. If a task modifies a variable in the variables map, or add a new variable, all dependent tasks will have access to this modification.
+
+For example:
+
+[source, groovy]
+----
+// task1
+String var = variables.get("one_variable")
+variables.put("one_variable", "foo")
+variables.put("other_variable", "bar")
+----
+
+[source, groovy]
+----
+// task2 depends on task1
+println variables.get("one_variable") // will display "foo"
+println variables.get("other_variable") // will display "bar"
+----
+
+If a task depends on several tasks and each task modifies the same variable, the final value of the variable
+which is propagated down the execution chain, depends on the *order of task execution*.
+Therefore, the users need to take appropriate measures to prevent any undesired effects of such race conditions.
+
+You can set any link:http://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html[Java serializable object, window="_blank"]
+ as a variable value. They will be converted into strings using
+http://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#toString()[toString()^]
+ method when required, for instance, to make those values available as environment variables in native tasks.
+
+
+Sometimes we want to propagate values between tasks and still define a default value.
+Setting the inherited field to true a task variable will read its value from a the *variables* map.
+Below a XML schema that shows two tasks: *first* and *second*. Task *first* insert a new
+variable using `variables.put("inherited_var", "somevalue")`. Task *second* declares *inherited_var*
+but with *defaultvalue*, this value will be overwritten by the  *first* task, `variables.put("inherited_var", "somevalue")`.
+This might be useful if for some reason the *first* task fails before inserting the *inhereted_var* value on *variables*
+map.
+
+[source, xml]
+----
+<task name="first" >
+    <scriptExecutable>
+        <script>
+            <code language="groovy">
+                <![CDATA[
+                variables.put("inherited_var", "somevalue")
+                ]]>
+            </code>
+        </script>
+    </scriptExecutable>
+</task>
+...
+<task name="second">
+    <variables>
+        <variable name="inherited_var" value="defaultvalue" inherited="true"/>
+    </variables>
+    <depends>
+        <task ref="before"/>
+    </depends>
+...
+</task>
+----
+
+
+===== Dynamic replacements
+
+It is possible to combine recursive definitions of variables.
```

Comment:
You should make a reminder of variable replacement. e.g. We've seen in paragraph... that we can use the ${} syntax to reuse variables, etc

Reply:
Before talking of the complex concept of recursive dynamic replacement, you should introduce the concept of dynamic replacement

Reply:
ok